go-controller/pkg/ovn/egressip.go

```diff
@@ -865,27 +866,27 @@ func (e *EgressIPController) deleteEgressIPAssignments(name string, statusesToRe
 				continue
 			}
 			podNamespace, podName := getPodNamespaceAndNameFromKey(podKey)
-			ni, err := e.networkManager.GetActiveNetworkForNamespace(podNamespace)
-			if err != nil {
-				return fmt.Errorf("failed to get active network for namespace %s", podNamespace)
-			}
 			cachedNetwork := e.getNetworkFromPodAssignment(podKey)
-			err = e.nodeZoneState.DoWithLock(statusToRemove.Node, func(key string) error {
+			if cachedNetwork == nil {
```

Comment:
hmm for deletion of EIP status we rely on the NAD being present in that namespace that determines the active network? that does sound wrong cause the NAD / namespace could just be gone i guess in order of events we don't control...
the fix looks correct to me but the fact that not a single unit test had to be changed worries me :D 
@martinkennelly can we track this somewhere in TODOs as we need to add a test for the sequence of events of deletion if a NAD doesn't exist and ensure nothing breaks... 

meanwhile @trozet why was GetActiveNetworkForNamespace not working in this specific instances? I assume this is some CI issue you happened to fix in this PR right?

Reply:
well the tests do fail now, thast how i caught it :) It's because the old getActiveNetworkForNamespace behavior would assume default network if a NAD didn't exist. Now it wont do that anymore if the namespace has the label. So an error was getting thrown here during the test and the status was not being removed in the e2es, causing them to fail. I'm not sure if the false positive was just enough to clean up stuff correctly for egress IP. Will defer to @martinkennelly for that.